src/components/views/dialogs/ChatCreateOrReuseDialog.js

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2017 Vector Creations Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import sdk from '../../../index';
+import dis from '../../../dispatcher';
+import MatrixClientPeg from '../../../MatrixClientPeg';
+import DMRoomMap from '../../../utils/DMRoomMap';
+import AccessibleButton from '../elements/AccessibleButton';
+import Unread from '../../../Unread';
+import classNames from 'classnames';
+import createRoom from '../../../createRoom';
+
+export default class CreateOrReuseChatDialog extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this._onNewDMClick = this._onNewDMClick.bind(this);
+        this.dispatcherRef = dis.register(this._onAction.bind(this));
+    }
+
+    componentWillUnmount() {
+        dis.unregister(this.dispatcherRef);
+    }
+
+    _onAction(payload) {
+        switch(payload.action) {
+            case 'view_room':
+                this.props.onFinished(true);
```

Comment:
>seems entirely possible that something else will decide to view a different room while the modal is being displayed

This is true. I would feel better about using a callback if that was the only thing that RoomTile was responsible for. i.e. move `view_room` dispatch out of RoomTile and into the `onClick` that gets passed into it.